### PR TITLE
Make EqualityTestGuard a default taint sanitizer guard

### PR DESCRIPTION
**Pull request by owen-mc**
Adds the ability to create a taint sanitizer guard that applies in all configurations. (This is already possible for taint sanitizers.)

Also takes EqualityTestGuard from `Xss.qll` and makes it a default taint sanitizer guard. The thinking is that for most configurations, being tainted means being user controlled, and comparing a value to a constant means that it has been checked and found to be an expected value.

### CODE REVIEW

File: ql/src/semmle/go/dataflow/internal/TaintTrackingUtil.qll [hunk 1]
```diff
@@ -186,4 +186,28 @@ abstract class DefaultTaintSanitizer extends DataFlow::Node { }
...
+/**
+ * Holds if `guard` should be a sanitizer guard in all global taint flow configurations
+ * but not in local taint.
+ */
+predicate isDefaultTaintSanitizerGuard(DataFlow::BarrierGuard guard) {
+  guard instanceof DefaultTaintSanitizerGuard
+}
+
+/**
+ * A check against a constant value, considered a barrier for XSS.
```
Comment (smowton) [inline ql/src/semmle/go/dataflow/internal/TaintTrackingUtil.qll:205]: Comment out of date

**Review by smowton — APPROVED** [review]
Looks good -- let's see if dist-compare shows anything interesting

**Comment by smowton** [comment]
Oh also, if merged this will warrant a change note

**Comment by sauyon** [comment]
How does one disable this if it's enabled by default?

**Comment by owen-mc** [comment]
I just noticed that in the comment for `BarrierGuard` it says

> It is important that all extending classes in scope are disjoint.

So before merging this we should look into existing barrier guards which are not disjoint from it and see if we can make them disjoint somehow.

**Comment by owen-mc** [comment]
I've updated this to exclude comparisons to `nil`. This should solve the performance problems that were occurring in repos with lots of nil checks. It shouldn't reduce the benefit because it seems highly unlikely that security-sensitive operations would be done on values which have been determined to be nil. I suppose there might be an exception for queries specifically looking for the use of `nil` somewhere.

**Pull request merged**